src/test/java/fi/otavanopisto/kuntaapi/test/utils/PtvDownload.java

```diff
@@ -0,0 +1,218 @@
+    List<String> ids = new ArrayList<>();
+    for (File file : files) {
+      ServiceChannelTypeExtract serviceChannelTypeExtract = getObjectMapper().readValue(file, ServiceChannelTypeExtract.class);
+      if (resolveChannelType(serviceChannelTypeExtract.getServiceChannelType()) == type) {
+        ids.add(String.format("      \"%s\"",  StringUtils.stripEnd(file.getName(), ".json")));
+      }
+    }
+    
+    System.out.println(String.format("\n    public final static String[] %s_SERVICE_CHANNELS = {", type.name()));
+    System.out.println(StringUtils.join(ids, ",\n"));
+    System.out.println("    };");
+  }
+
+  private PtvDownload downloadOrganizations() throws JsonParseException, JsonMappingException, IOException {
+    File[] organizationFiles = listJsonFiles(getOrganizationsFolder());
+    for (File organizationFile : organizationFiles) {
+      downloadOrganizationServices(organizationFile);
+    }
+    
+    return this;
+  }
+  
+  private File[] listJsonFiles(File folder) {
+    return folder.listFiles(new FilenameFilter() {
+      @Override
+      public boolean accept(File dir, String name) {
+        return name.endsWith(".json");
+      }
+      
+    });  
+  }
+  
+  private File getOrganizationsFolder() {
+    return new File(getPtvFolder(), "organizations");
+  }
+
+  private File getServicesFolder() {
+    return new File(getPtvFolder(), "services");
+  }
+
+  private File getServiceChannelsFolder() {
+    return new File(getPtvFolder(), "servicechannels");
+  }
+  
+  private File getPtvFolder() {
+    return new File(System.getProperty("user.dir"), "src/test/resources/ptv");
+  }
+  
+  private void downloadOrganizationServices(File organizationFile) throws IOException, JsonParseException, JsonMappingException {
+    V7VmOpenApiOrganization organization = getObjectMapper().readValue(organizationFile, V7VmOpenApiOrganization.class);
+    List<V5VmOpenApiOrganizationService> services = organization.getServices();
+    for (V5VmOpenApiOrganizationService service : services) {
+      downloadService(service);
+    }
+  }
+  
+  private ObjectMapper getObjectMapper() {
+    return new ObjectMapper();
+  }
+
+  private void downloadService(V5VmOpenApiOrganizationService service) throws JsonParseException, JsonMappingException, IOException {
+    String serviceId = service.getService().getId().toString();
+    File servicesFolder = getServicesFolder();
+    
+    String url = String.format("https://api.palvelutietovaranto.trn.suomi.fi/api/%s/Service/%s", PTV, serviceId);
+    File target = new File(servicesFolder, String.format("%s.json", serviceId));
+    if (!target.exists()) {
+      try {
+        download(url, target);          
+        System.out.println(url);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+    
+    downloadServiceChannels(target);
+  }
+  
+  private void downloadServiceChannels(File serviceFile) throws JsonParseException, JsonMappingException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)